Dockerfile-pytorch-ngc

```diff
@@ -0,0 +1,40 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+# NGC images contain user owned files in /usr/lib
+RUN chown root:root /usr/lib
+
+# Copy various shell scripts that group dependencies for install
+COPY dockerfile_scripts /tmp/det_dockerfile_scripts
+
+RUN /tmp/det_dockerfile_scripts/install_deb_packages.sh
+
+# We uninstall these packages after installing. This ensures that we can
+# successfully install these packages into containers running as non-root.
+# `pip` does not uninstall dependencies, so we still have all the dependencies
+# installed.
+RUN python -m pip install determined && python -m pip uninstall -y determined
+
+RUN python -m pip install -r /tmp/det_dockerfile_scripts/additional-requirements-torch.txt \
+    -r /tmp/det_dockerfile_scripts/additional-requirements.txt \
+    -r /tmp/det_dockerfile_scripts/notebook-requirements.txt
+
+# Install jupyter notebook requirements
+RUN python -m pip install -r /tmp/det_dockerfile_scripts/notebook-requirements.txt
+ENV JUPYTER_CONFIG_DIR=/run/determined/jupyter/config
+ENV JUPYTER_DATA_DIR=/run/determined/jupyter/data
+ENV JUPYTER_RUNTIME_DIR=/run/determined/jupyter/runtime
+
+RUN /tmp/det_dockerfile_scripts/add_det_nobody_user.sh
```

Comment:
move the add_det_nobody_user.sh to be next to the install_libnss_determined.sh.  They're related goals.

Right now, add_det_nobody_user.sh is kind of in the middle of a bunch of python-related installations, so the order doesn't really make sense.

Arguably, these two scripts should happen earlier, right after install_deb_packages.sh, because these are like "system-level" operations while the pip ones are more like "virtual-env-level operations"